api/python/provisioner/commands/check.py

```diff
@@ -269,6 +269,42 @@ def make_decision(self, check_result: CheckResult):
         :return:
         """
 
+class DeploymentDecisionMaker(DecisionMaker):
+
+    """Class analyses `CheckResult` structure and will decide
+       to continue or to stop Factory Deployment routine
+    """
+
+    def _check_critical_errors(self, check_result: CheckResult):
```

Comment:
@divya-vijay  why do we need that copy-paste here?
@dmitrydb As long as the method is not used  I think we either need to move it to the base class or remove at all. I believe it (`ust as example that `DecisionMaker` child classes can threat some
        errors as critical to raise appropriate Exceptio`) is a kind of clear thing and don't need to be emphasized in a such a way.

Reply:
We're looking at making this implementation better by handling critical checks in DM class. 
Since for GA, we had decided to let the user run CLI for post-checks, this was not needed. Now, working on handling the checks automatically, so this might be needed next.